File: ai-community-backend/src/main/java/com/aicommunity/entity/Activity.java
package com.aicommunity.entity;

import lombok.Data;
import java.io.Serializable;
import java.util.Date;

/**
 * 活动实体类
 *
 * @author AI Community Team
 */
@Data
public class Activity implements Serializable {
    private static final long serialVersionUID = 1L;

    /**
     * 活动ID
     */
    private Long id;

    /**
     * 活动标题
     */
    private String title;

    /**
     * 活动内容（HTML格式）
     */
    private String content;

    /**
     * 封面图URL
     */
    private String cover;

    /**
     * 工具ID（-1表示扶摇Agent应用）
     */
    private Long toolId;

    /**
     * 工具名称
     */
    private String toolName;

    /**
     * 活动类型：training-培训，competition-竞赛，sharing-分享
     */
    private String type;

    /**
     * 活动日期
     */
    private Date date;

    /**
     * 发布者ID
     */
    private Long authorId;

    /**
     * 发布者姓名
     */
    private String authorName;

    /**
     * 发布者头像
     */
    private String authorAvatar;

    /**
     * 已报名人数
     */
    private Integer registeredCount;

    /**
     * 活动状态：upcoming-即将开始，ongoing-进行中，ended-已结束
     */
    private String status;

    /**
     * 当前用户是否已报名
     */
    private Boolean isRegistered;

    /**
     * 当前用户是否是发布者
     */
    private Boolean isAuthor;

    /**
     * 是否可以编辑
     */
    private Boolean canEdit;

    /**
     * 是否可以删除
     */
    private Boolean canDelete;

    /**
     * 创建时间
     */
    private Date createTime;

    /**
     * 更新时间
     */
    private Date updateTime;
}

File: frontend/src/api/activity.ts
// 活动相关接口
import request from './request'

// 活动接口
export interface Activity {
  id: number
  title: string
  content: string
  cover?: string
  toolId?: number
  toolName?: string
  type: string
  date: string | Date
  authorId: number
  authorName: string
  authorAvatar?: string
  registeredCount?: number
  status?: string
  isRegistered?: boolean
  isAuthor?: boolean
  canEdit?: boolean
  canDelete?: boolean
  createTime?: string | Date
  updateTime?: string | Date
}

// 分页响应
export interface PageResult<T> {
  list: T[]
  total: number
  page: number
  pageSize: number
}

// 创建活动请求
export interface ActivityCreateRequest {
  title: string
  toolId?: number
  type: string
  date: string
  cover: string
  content: string
}

// 更新活动请求
export interface ActivityUpdateRequest {
  title: string
  toolId?: number
  type: string
  date: string
  cover: string
  content: string
}

// 活动报名响应
export interface ActivityRegistrationResponse {
  registered: boolean
  registeredCount: number
  registrationUser?: {
    id: number
    name: string
    avatar: string
    registeredAt: string
  }
}

// 获取活动列表
export const getActivities = async (params?: {
  toolId?: number
  type?: string
  status?: string
  page?: number
  pageSize?: number
}): Promise<PageResult<Activity>> => {
  return await request.get<PageResult<Activity>>('/activities', { params })
}

// 获取活动详情
export const getActivityDetail = async (id: number): Promise<Activity> => {
  return await request.get<Activity>(`/activities/${id}`)
}

// 创建活动
export const createActivity = async (data: ActivityCreateRequest): Promise<Activity> => {
  return await request.post<Activity>('/activities', data)
}

// 更新活动
export const updateActivity = async (id: number, data: ActivityUpdateRequest): Promise<Activity> => {
  return await request.put<Activity>(`/activities/${id}`, data)
}

// 删除活动
export const deleteActivity = async (id: number): Promise<void> => {
  await request.delete(`/activities/${id}`)
}

// 报名活动
export const registerActivity = async (id: number): Promise<ActivityRegistrationResponse> => {
  return await request.post<ActivityRegistrationResponse>(`/activities/${id}/register`)
}

// 取消报名
export const cancelRegistration = async (id: number): Promise<ActivityRegistrationResponse> => {
  return await request.delete<ActivityRegistrationResponse>(`/activities/${id}/register`)
}

// 获取报名用户列表
export const getRegistrations = async (id: number, params?: {
  page?: number
  pageSize?: number
}): Promise<PageResult<ActivityRegistrationResponse['registrationUser']>> => {
  return await request.get<PageResult<ActivityRegistrationResponse['registrationUser']>>(`/activities/${id}/registrations`, { params })
}
